src/test/compile-fail/auxiliary/dropck_eyepatch_extern_crate.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(generic_param_attrs)]
+#![feature(dropck_eyepatch)]
+
+// The point of this test is to illustrate that the `#[may_dangle]`
```

Comment:
Funny to have this comment in the `auxiliary` crate, and not the main test.

Also, why not make a `ui/dropck_eyepatch/` directory and collect tests there, and then have `ui/dropck_eyepatch/auxiliary`? (Could also be `compile-fail`, though I would like to move everything to `ui` at some point.)